tests/functional/tests/backup.py

```diff
@@ -17,21 +16,27 @@
 class BackupTest(unittest.TestCase):
     """Code for backup test."""
 
+    def setUp(self) -> None:
+        zaza.get_or_create_libjuju_thread()
+        model_name = zaza.model.get_juju_model()
+        self.model = COUModel(model_name)
+
+    def tearDown(self) -> None:
+        zaza.sync_wrapper(self.model._model.disconnect)()
+        zaza.clean_up_libjuju_thread()
+
     def test_backup(self):
         """Backup Test."""
         # create the COU_DATA path to place the backup file
-        pathlib.Path(COU_DATA).mkdir(parents=True, exist_ok=True)
-        zaza.get_or_create_libjuju_thread()
+        COU_DATA.mkdir(parents=True, exist_ok=True)
         sync_backup = zaza.sync_wrapper(backup)
 
         logger.info("Running backup test....")
-        model_name = zazamodel.get_juju_model()
-        model = COUModel(model_name)
-        sync_connect = zaza.sync_wrapper(model.connect)
-
-        sync_connect()
-        backup_file = sync_backup(model)
-        logger.info("Backup file: %s", backup_file)
-        assert os.path.getsize(backup_file) > 0
-        self.addCleanup(os.remove, backup_file)
-        self.addCleanup(zaza.clean_up_libjuju_thread)
+        # NOTE(gabrielcocenza) mocking _check_db_relations to return True allows
+        # to test that backup without the need of deploying a heavy OpenStack bundle.
+        with patch("cou.steps.backup._check_db_relations") as mock_relations:
```

Comment:
I would prefer to run backup with snap, but I see that it would not be possible.